clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -253,7 +253,12 @@ public boolean rename(Path src, Path dst) throws IOException {
         // Throws FileNotFoundException when src does not exist. mimics s3a's behaviour in
         // https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L1505
         LakeFSFileStatus srcStatus;
-        srcStatus = getFileStatus(src);
+        try {
+            srcStatus = getFileStatus(src);
+        }
+        catch (FileNotFoundException e) {
+            return false;
+        }
```

Comment:
This explicitly changes the behaviour documented on 253-254.  Is this intentional?  (IIRC `git blame` will show that the old behaviour is actually required by something S3A-ish, so we should see what S3A does for its contract).

Reply:
Reverted